app/commands/v2/publish.rb

```diff
@@ -39,6 +39,7 @@ def call
         clear_published_items_of_same_locale_and_base_path(content_item, translation, location)
 
         set_public_updated_at(content_item, previous_item, update_type)
+        set_first_published_at(content_item)
```

Comment:
We do a lot of stuff when we publish.  Might be worth another refactor later, but I'm not sure how much it can be reduced down while still being clear.